src/pydvl/influence/general.py

```diff
@@ -63,25 +59,30 @@ def compute_influence_factors(
     :returns: An array of size (N, D) containing the influence factors for each
         dimension (D) and test sample (N).
     """
-    test_grads = zero_tensor(
-        shape=(len(test_data.dataset), model.num_params),
-        dtype=test_data.dataset[0][0].dtype,
-        device=model.device,
+    tensor_util: Type[TensorUtilities] = TensorUtilities.from_twice_differentiable(
+        model
     )
-    for batch_idx, (x_test, y_test) in enumerate(
-        maybe_progress(test_data, progress, desc="Batch Test Gradients")
+    stack = tensor_util.stack
+    unsqueeze = tensor_util.unsqueeze
+    cat = tensor_util.cat
+
+    test_grads = []
```

Comment:
appending tensors to a list and then stacking is typically slower and uses more memory than just filling a previously create one. Any reason to go back to the old setup in all these methods?

Reply:
@Xuzzo I would like to postpone this optimization question, as abstracting away the bracket access is a further complexity.

Reply:
@Xuzzo I would like to postpone this optimization question, as abstracting away the bracket access is a further complexity.

Reply:
@Xuzzo I would like to postpone this optimization question, as abstracting away the bracket access is a further complexity.

Reply:
@Xuzzo pre-allocating result tensor is implemented